File: frontend/src/types/room.ts
export interface Room {
  id: number
  number: string
  floor: string
  type: string
  status: string
  price: number
  createdAt: string
  updatedAt: string
}

export interface RoomRequest {
  number: string
  floor: string
  type: string
  status: string
  price: number
}

export interface RoomResponse {
  id: number
  number: string
  floor: string
  type: string
  status: string
  price: number
  createdAt: string
  updatedAt: string
}

export interface RoomListResponse {
  rooms: RoomResponse[]
  total: number
  page: number
  size: number
  totalPages: number
}

export interface RoomQueryParams {
  page?: number
  size?: number
  number?: string
  floor?: string
  status?: string
}

export interface ApiResponse<T = any> {
  code: number
  message: string
  data: T
}

File: src/main/java/com/hotel/entity/Room.java
package com.hotel.entity;

import lombok.*;
import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "rooms")
@NoArgsConstructor
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false, length = 10)
    private String number;

    @Column(nullable = false, length = 10)
    private String floor;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RoomType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RoomStatus status;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer capacity;

    @Column(columnDefinition = "TEXT")
    private String images;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
